nomos-services/network/src/backends/libp2p.rs

```diff
@@ -84,22 +100,43 @@ impl NetworkBackend for Libp2p {
     type NetworkEvent = Event;
 
     fn new(config: Self::Settings, overwatch_handle: OverwatchHandle) -> Self {
-        let mut mixnet_client = MixnetClient::new(config.mixnet_client, OsRng);
+        let mixnet_client = MixnetClient::new(config.mixnet_client.clone(), OsRng);
         let (commands_tx, mut commands_rx) = tokio::sync::mpsc::channel(BUFFER_SIZE);
         let (events_tx, _) = tokio::sync::broadcast::channel(BUFFER_SIZE);
-        let libp2p = Self {
-            events_tx: events_tx.clone(),
-            commands_tx,
-        };
-        overwatch_handle.runtime().spawn(async move {
-            use tokio_stream::StreamExt;
 
+        let cmd_tx = commands_tx.clone();
+        overwatch_handle.runtime().spawn(async move {
             let Ok(mut stream) = mixnet_client.run().await else {
                 tracing::error!("Could not quickstart mixnet stream");
                 return;
             };
 
+            while let Some(result) = stream.next().await {
+                match result {
+                    Ok(msg) => {
+                        tracing::debug!("receiving message from mixnet client");
+                        let Ok(MixnetMessage { topic, message }) = serde_json::from_slice(&msg) else {
+                            tracing::error!("failed to deserialize json received from mixnet client");
+                            continue;
+                        };
+
+                        cmd_tx.send(
+                            Command::DirectBroadcastAndRetry { topic, message, retry: 0 })
+                                .await
+                                .unwrap_or_else(|_| tracing::error!("could not schedule broadcast")
+                        );
+                    },
+                    Err(e) => {
+                        todo!("Handle mixclient error: {e}");
+                    }
+                }
+            }
+        });
```

Comment:
Just a short question. Does it also work if we move this under the `tokio::select!` in the other async task, insteading of spawning this separated task? I'm happy with having this task for consuming the stream little faster, but asking to check if I understand it correctly.

Reply:
It's not a matter of consuming the little stream faster, is more of a matter of separation of concerns. 
Yes, a single select would probably work (unless there are deadlocks somewhere which I haven't checked for, after all it's just a single threaded executor), but there's not reason to have it all in one place IMO. The mixnet is separated from gossipsub and I think it clearer if we separate those in the implementation.

Reply:
Clear. Thank you.